src/messages.h.jinja

```diff
@@ -0,0 +1,45 @@
+// NOTE: This file has been generated automatically by “update-message-registry.py”.
+//       Do not edit manually!
+
+#ifndef MESSAGES_H
+#define MESSAGES_H
+
+#include <stdint.h>
+
+/**
+ * @name Indexes of the log messages
+ *
+ * @added 1.6.0
+ *
+ */
+enum xkb_message_index {
+    {% for entry in entries %}
+    /** {{ entry.description }} */
+    {{ entry.message_index }},
+    {% endfor %}
+    XKB_LOG_MESSAGE_MAX_INDEX
+};
+
+typedef uint32_t xkb_message_code_t;
+
+#define XKB_INVALID_MESSAGE_INDEX 0xFFFFFFFFu
+
+typedef struct {
+    xkb_message_code_t code;
+    const char *identifier;
+    const char *description;
+} xkb_message_entry;
+
+xkb_message_code_t
+xkb_message_code(enum xkb_message_index message_index);
```

Comment:
If it were me, I'd have just one function, `const *struct xkb_message_entry xkb_message_get(xkb_message_code_t code)`, and use it everywhere by accessing the fields, e.g. `xkb_message_get(...).identifier` instead of `xkb_message_identifier(...)`.

But that's a matter of taste, take it or leave it.